### DISCUSSION TITLE
Sloshing modelling

### DISCUSSION THREAD

**Discussion by samahiz** [rubensamarojr/polymps#9] [Q&A]
Hello,
I would like to model sloshing within a closed container. 
I notice that the mesh motion is stated to not work. Is it possible to set a time-dependent acceleration?

**Comment by rubensamarojr** [comment]
Hi,

I have updated the code with an example of 3D sloshing uisng polygon wall.
The json file is https://github.com/rubensamarojr/polymps/blob/master/input/sloshing_3D_bulian_lo0p00930.json
The forced motion must be defined in a hard-coded way, in the function below https://github.com/rubensamarojr/polymps/blob/02d1811e36eb76d16df219851d36da4baeb5c7cc/src/PolygonMesh.cpp#L174
In this function, the motion is transfered to the mesh as node displacements (dx, dy, dz).
There are some examples of constant and sinusoidal motions in this function. For now, you have to write a function for the motion velocity. But with few changes, it is possible to read an external file and set the velocity and the displacements.

After modifying the function, don't forget to build the code again, i.e., run the command

```
make all
```

Feel free to ask me if you need any help.
